lib/facility.rb

```diff
@@ -1,14 +1,61 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :collected_fees, :registered_vehicles
 
-  def initialize(name, address, phone)
+  def initialize(name:, address:, phone:)
     @name = name
     @address = address
     @phone = phone
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def administer_written_test(person)
+    if @services.include?("Written Test") && person.permit? && person.age >= 16
+      person.license_data[:written] = true
+      true
+    else
+      false
+    end
+  end
+
+  def administer_road_test(person)
+    if person.license_data[:written] == true && @services.include?("Road Test")
+      person.license_data[:license] = true
+      true
+    else
+      false
+    end
+  end
+
+  def renew_drivers_license(person)
+    if @services.include?("Renew License") && person.license_data[:license] == true
+      person.license_data[:renewed] = true
+      true
+    else
+      false
+    end
+  end
+
+  def register_vehicle(car)
+    if @services.include?("Vehicle Registration")
+      @registered_vehicles << car
+      car.registration_date = Date.today
+      car.plate_type!
+      if car.plate_type == :regular
```

Comment:
In the future, consider moving this logic into its own helper method. Maybe it's a method called `collect_fees` or something, that the register vehicle method could call. When thinking about Single Responsibility Principle, we want each method to do one thing. Currently, register_vehicle is registering a vehicle, and changing the plate, and collecting fees. If we can break these sections of logic into their own helper methods that register_vehicle calls, then we've separated out the responsibilities.